### ISSUE TITLE
Default Buffer rule when no internal procedures

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#363]
When there are no internal procedures, the default buffer rule should not complain.

````
FIND customer.
````

Will trigger the rule - but the rule makes no sense if there are no internal procedures which can cause buffer confusion.

**Comment by movedoa** [comment]
I am not a fan of this.
It is always best practive to define your own buffers.
Stuff like this will bite you when you try to refactor this code.

I see no downside of defininig buffers, only potential benefits.

**Comment by stefandrissen** [comment]
Adding buffers to a program that contains no internal procedures is code noise. Which I then have to read.

The buffer bite will be caught by this rules when refactored.

**Comment by gquerret** [comment]
My personal preference is to have a `define buffer xxx for xxx` everywhere, even when there's no internal procedure. So it might be an option at best.
Rule UnusedBuffer (issue #315) has been added recently in case you're afraid of defining too many buffers.

**Comment by gquerret** [comment]
Property added in develop branch
